src/javaharness/java/arcs/webimpl/DeviceClientJsImpl.java

```diff
@@ -20,15 +22,16 @@
 
     private PortableJsonParser jsonParser;
     private Map<String, ArcsEnvironment.SuggestionListener> inProgress;
-    private PECInnerPort port;
+    private PECInnerPortFactory portFactory;
```

Comment:
As an aside, in general, fields that are only ever assigned in constructors (e.g. const) should be marked final (e.g. private final PECInnerPortFactory portFactory)

Reply:
done (and actually found one redundant data member that i removed in PEC port impl)